js/game.js

```diff
@@ -0,0 +1,77 @@
+/** @type {HTMLCanvasElement} */
+
+class Game {
+  constructor(ctx, width, height, player) {
+    this.ctx = ctx;
+    this.width = width;
+    this.height = height;
+    this.car = car;
+    this.intervalId = null;
+    this.frames = 0;
+    this.roadblock = [];
+  }
+
+  start() {
+    this.intervalId = setInterval(this.update, 1000 / 60);
+  }
+  //Update needs to be an arrow function because "this" needs to refer to the class and not the update method
+  update = () => {
+    //Game logic here
+    this.frames++;
+    this.clear();
+    this.car.newPos();
+    this.car.draw();
+    this.updateRoadBlocks();
+    this.checkGameOver();
+  };
+
+  stop() {
+    clearInterval(this.intervalId);
+  }
+
+  clear() {
+    this.ctx.clearRect(0, 0, this.width, this.height);
+  }
+
+  updateRoadBlocks() {
+    for (let i = 0; i < this.roadblock.length; i++) {
+      this.roadblock[i].y += 2;
+      this.roadblock[i].draw();
+    }
+    //the if statement is to create the roadblocks
+    //which we only want to do every 120 frames (2 seconds)
+    if (this.frames % 60 === 0) {
+      let randomSize = Math.floor(Math.random() * 150 - 10) + 10; //roadblock will appear in random sizes
+      let randomX =
+        Math.floor(Math.random() * this.width - randomSize) + randomSize;
+      this.roadblock.push(
+        new Enemy(randomX, 0, randomSize, 100, "red", this.ctx)
```

Comment:
Adjust the `height` and `width` that you are passing to each roadBlock.
Probably the `height` could be static and adapt the `randomSize` to the `width` of the roadBlocks coming.